src/OpenTelemetry.Contrib.Instrumentation.Remoting/Implementation/TelemetryDynamicSink.cs

```diff
@@ -0,0 +1,232 @@
+            if (bCliSide)
+            {
+                // Start new outgoing activity
+                var act = RemotingActivitySource.StartActivity(ActivityOutName, ActivityKind.Client);
+                if (act != null)
+                {
+                    if (act.IsAllDataRequested && reqMsg is IMethodMessage methodMsg)
+                    {
+                        SetStartingActivityAttributes(act, methodMsg);
+                    }
+
+                    var callContext = (LogicalCallContext)reqMsg.Properties["__CallContext"];
+
+                    this.options.Propagator.Inject(new PropagationContext(act.Context, Baggage.Current), callContext, InjectActivityProperties);
+                }
+            }
+            else
+            {
+                // We are on server, need to start new or attach to an existing incoming activity.
+
+                var callContext = (LogicalCallContext)reqMsg.Properties["__CallContext"];
+                var activityParentContext = this.options.Propagator.Extract(default, callContext, ExtractActivityProperties);
+
+                Activity ourActivity;
+
+                // Do we already have an incoming activity?
+                // Existing activity might be started by an instrumentation layer higher up the
+                // stack. E.g. if we are using http as a transport for remoting, and we have
+                // instrumented incoming http request, then that instrumentation happens before
+                // our "ProcessMessageStart" and we just need to attach to that existing activity.
+                var parentActivity = Activity.Current;
+                if (parentActivity != null)
+                {
+                    ourActivity = RemotingActivitySource.StartActivity(ActivityInName, ActivityKind.Server);
+                }
+                else
+                {
+                    // We don't have an existing incoming activity. Start a brand new one ourselves.
+
+                    ourActivity = RemotingActivitySource.StartActivity(ActivityInName, ActivityKind.Server, activityParentContext.ActivityContext);
+                }
+
+                if (ourActivity != null && ourActivity.IsAllDataRequested && reqMsg is IMethodMessage methodMsg)
+                {
+                    SetStartingActivityAttributes(ourActivity, methodMsg);
+                }
+            }
+        }
+
+        public void ProcessMessageFinish(IMessage replyMsg, bool bCliSide, bool bAsync)
+        {
+            // This will be called:
+            // 1) On the server after we are done processing the call and are returning back to client
+            //    => Current activity should be the "ActivityIn", we need to stop it
+            //
+            // 2) On the client, when the server call returns
+            //    => Current activity should be the "ActivityOut", we need to stop it as well
+
+            try
+            {
+                if (this.options.Filter?.Invoke(replyMsg) == false)
```

Comment:
do we have to run the Filter in `ProcessMessageFinish` as well? If this was filtered in `ProcessMessageStart`, then we wouldn't have started a new activity. So activity will be null, or someone else's activity, and this case seems handled already in the below codes..

Reply:
Hmm, not sure about this one.

If we _do_ run the filter then we know that we except our activity back at this point. Is it worth logging that we got someone else's (or null) instead?

This is related to the other thread about `aServerActivity`. In scenarios like below, when someone doesn't stop their activity or stops ours - should we log such things in general? Or is that too noisy and better to just continue silently?
```csharp
// Scenario 1:
{
    ProcessMessageStart
    {
        RemotingActivitySource.StartActivity("RequestIn");
    }

    // If user code starts activity and doesn't stop it:
    SomeOtherActivitySource.StartActivity("user-operation")

    ProcessMessageFinish
    {
        if (Activity.Current?.OperationName != "RequestIn")
        {
            // Log Warning
        }
    }
}

// Scenario 2:

{
    ProcessMessageStart
    {
        RemotingActivitySource.StartActivity("RequestIn");
    }

    // Or if user code stops "our" activity
    Activity.Current?.Stop();

    ProcessMessageFinish
    {
        if (Activity.Current?.OperationName != "RequestIn")
        {
            // LogWarning
        }
    }
}
```

Reply:
Logging is done to EventSource. I think we should log warning level log message. If somethings doesn't work as expected, then those logs are the only option for us to investigate. (assuming we cannot attach debugger)

Reply:
I looked at how this was done in other instrumentations ([ASP.NET Core](https://github.com/open-telemetry/opentelemetry-dotnet/blob/6633dc846e20c72460ae478f86e7225e32110717/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L143), [ASP.NET](https://github.com/open-telemetry/opentelemetry-dotnet/blob/6633dc846e20c72460ae478f86e7225e32110717/src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs#L201)) and they do simply silently continue when they encounter an activity that they didn't start.

I changed the Remoting one to be the same. It'll only log an error now if something went wrong processing "our" activity.